Products/CMFPlone/browser/icon.py

```diff
@@ -0,0 +1,60 @@
+# -*- coding: utf-8 -*-
+from OFS.Image import File
+from plone.registry.interfaces import IRegistry
+from Products.CMFCore.interfaces import ISiteRoot
+from zope.component import adapter
+from zope.component import getUtility
+from zope.component.hooks import getSite
+from zope.interface import implementer
+from zope.interface import Interface
+from zope.location.interfaces import LocationError
+from zope.traversing.interfaces import ITraversable
+from zope.publisher.interfaces import IPublishTraverse
+
+
+@implementer(IPublishTraverse)
+class IconView:
+
+    prefix = 'plone.staticresources.icon.'
+
+    def publishTraverse(self, request, name):
+        request['TraversalRequestNameStack'] = []
+        return self
+
+    def __init__(self, context, request):
```

Comment:
Instead of redefining the `__init__` of the view I would set the icon attribute in the `publishTraverse` method.

Reply:
... and inherit from BrowserView

Reply:
Setting the icon attribute in publishTraverse doesn't work for me. I just followed https://docs.plone.org/develop/plone/serving/traversing.html#custom-traversal.

Reply:
I used this several times. How does it not work?